aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -593,12 +593,9 @@ def presubmit(self, folder: Folder) -> CalcInfo:
                 raise PluginInternalError('CalcInfo should have the information of the code to be launched')
             this_code = load_node(code_info.code_uuid, sub_classes=(Code,))
 
-            this_withmpi = code_info.withmpi  # to decide better how to set the default
-            if this_withmpi is None:
-                if len(calc_info.codes_info) > 1:
-                    raise PluginInternalError('For more than one code, it is necessary to set withmpi in codes_info')
-                else:
-                    this_withmpi = self.node.get_option('withmpi')
+            this_withmpi = self.node.get_option('withmpi')  # set code mpi, default set to as calcjob one
+            if code_info.withmpi:
+                this_withmpi = code_info.withmpi
```

Comment:
So, this change actually makes more sense to me (why was previously the `withmpi` flag related to how many codes where provided before or even in what mode these where run? I mean, `withmpi` only refers to paralellization **within** each run of a code, not if these codes runs are then performed in parallel or serial). But just to mention that it would be good to also add some comment on this change on the commit log.

Reply:
> why was previously the withmpi flag related to how many codes where provided before or even in what mode these where run? I mean, withmpi only refers to paralellization within each run of a code, not if these codes runs are then performed in parallel or serial

Not exactly, for example if I have two codes, where the first is a pre-processing helper code must be run without mpi and the second one is the real simulation code better run with mpi. The user may only set `withmpi=True` as a flag in calcjob in intuitively without setting `withmpi=False` paticuliarly for the first hepler code. This will fail the running. 
I guess this is the reason why this default setting is not being properly set previously.

Reply:
Ahhhhh I understand now what the old code was doing. It was getting the `withmpi` flag from each code and if it didn't find it then it checked if there were one code, in which case it used the `withmpi` flag from the calcjob, or if there were many codes it just raised. It is a bit of a convoluted implementation though xD.

So, let me make sure I understand this: the `calcjob-withmpi` can be set **by the user** and is what is used by default on the codes, unless **the calcjob designer** specifically sets a `code-withmpi` which will override any behavior requested by the user. Is that correct?

Maybe it would be worth to make some info box explaining this in the [How-to sections for writing calcjobs](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/plugin_codes.html#preparing-for-submission) (not saying to do it now necessarily, we can add an issue for this, but do feel free if you are inspired to do so).